gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/reactive/handlers/api/processor/subscription/SubscriptionProcessor.java

```diff
@@ -166,15 +179,28 @@ private String computeCtxClientIdentifier(final MutableExecutionContext ctx, fin
 
     private String computeHashOrDefault(final String valueToHash, final String fallback) {
         try {
-            MessageDigest md = MessageDigest.getInstance("SHA-256");
-            md.update(valueToHash.getBytes());
-            byte[] digest = md.digest();
-            return Hex.toHexString(digest);
-        } catch (NoSuchAlgorithmException e) {
+            Cache<String, String> hashedValueCache = getOrCreateCache();
+            String hashedValue = hashedValueCache.get(valueToHash);
+            if (hashedValue == null) {
+                MessageDigest md = MessageDigest.getInstance("SHA-256");
+                md.update(valueToHash.getBytes());
+                byte[] digest = md.digest();
+                hashedValue = Hex.toHexString(digest);
```

Comment:
out of curiosity why not Base64 ?

Reply:
Because we don’t want the value to be readable